image_builder/notify.py

```
@@ -33,19 +33,33 @@ def __init__(self, send_notifications: bool = True):
             except KeyError as e:
                 raise ValueError(f"{e} environment variable must be set")
 
-    def post_build_progress(self, progress: Progress, codebase: Codebase):
+    def _prepare_message_blocks(self, codebase: Codebase, text_blocks=None):
+        if text_blocks is not None:
+            return text_blocks
+
+        if codebase is not None:
+            return {
+                "repository_name": codebase.revision.get_repository_name(),
+                "revision_commit": codebase.revision.commit,
+                "repository_url": codebase.revision.get_repository_url(),
+            }
+        return {}
```

What would be the expected behaviour `if text_blocks is not None and codebase is not None`?

Feels like it should maybe throw an exception?

---

I think so https://github.com/uktrade/ci-image-builder/pull/72#discussion_r1679635539 see my previous answer to Ant